pkg/client/monitoring/v1/types.go

```diff
@@ -22,10 +22,12 @@ import (
 )
 
 // Prometheus defines a Prometheus deployment.
+// +k8s:openapi-gen=true
 type Prometheus struct {
 	metav1.TypeMeta `json:",inline"`
 	// Standard object’s metadata. More info:
 	// https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#metadata
+	// +k8s:openapi-gen=false
```

Comment:
how come this is false?

Reply:
I had an issue with this(I ll try again and document it).  I assumed that the metadata field is already validated on every resources

Reply:
Ok so when you apply a resource with kubectl it adds a 'deletionTimestamp: ' field, but the spec requires a 'date' and doesn't allow 'nil', so the resource is rejected.